Services/Saml/classes/class.ilSamlAttributeMapping.php

```diff
@@ -0,0 +1,222 @@
+<?php
+// saml-patch: begin
+/* Copyright (c) 1998-2016 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+require_once 'Services/Saml/classes/class.ilSamlAttributeMappingRule.php';
+
+/**
+ * Class ilSamlAttributeMapping
+ * @author Michael Jansen <mjansen@databay.de>
+ */
+class ilSamlAttributeMapping implements ArrayAccess, Countable, Iterator
```

Comment:
would it be possible to extract AttributeMapping to a general abstract class. Would be helpful to use it in LDAP or Shibboleth, too.